### PR TITLE
PSE-10014: update cr connection response

### PR DESCRIPTION
**Pull request by Favouroked**
## Jira Ticket
PSE-10014
PSE-9631

## Description
Update cloud router connection response. Add `cloud_router_name`, `cloud_router_asn`, `bgp_state_list`. `bgp_state` field is deprecated.

### CODE REVIEW

**Comment by rjouhann** [comment]
The failed Test is expected as the change did not go through production yet.

File: internal/provider/datasource_cloud_conn.go [hunk 1]
```diff
@@ -353,6 +386,21 @@ func flattenCloudProvider(provider *packetfabric.AwsCloudProvider) []interface{}
 	return flattens
 }
 
+func flattenBgpStateList(bgp_state_list *[]packetfabric.BgpStateObj) []interface{} {
```
Comment (medzin) [inline internal/provider/datasource_cloud_conn.go:389]: The convention in Go is to use `mixedCaps` for everything, not `snake_case`. Please correct the name.

File: internal/provider/datasource_cloud_conn.go [hunk 2]
```diff
@@ -353,6 +386,21 @@ func flattenCloudProvider(provider *packetfabric.AwsCloudProvider) []interface{}
...
+	if bgp_state_list != nil {
+		flattens := make([]interface{}, len(*bgp_state_list), len(*bgp_state_list))
+
+		for i, bgp_state_obj := range *bgp_state_list {
```
Comment (medzin) [inline internal/provider/datasource_cloud_conn.go:393]: `mixedCaps` as well.

File: internal/packetfabric/cloud_router_connection.go [hunk 3]
```diff
@@ -74,6 +77,11 @@ type CloudRouterConnectionReadResponse struct {
 	Site                      string           `json:"site,omitempty"`
 }
 
+type BgpStateObj struct {
```
Comment (medzin) [inline internal/packetfabric/cloud_router_connection.go:80]: Please use `gofmt` for code formatting. All popular editors support it, and it is part of standard Go tooling.

**Review by medzin — CHANGES_REQUESTED** [review]

**Comment by rjouhann** [comment]
Not doing this anymore, see more details on the PSE-10053

**Comment by rjouhann** [comment]
@medzin I fixed the lint errors. The Test/Test is failing and I can't figure out why. Could you help?

**Comment by medzin** [comment]
`Test_ReadCloudRouterConnection` fails, but I cannot see why. I need some time to debug it locally.

File: internal/packetfabric/cloud_router_connection_test.go [hunk 4]
```diff
@@ -229,30 +242,38 @@ func buildMockCloudRouterReadResp(description string) []byte {
 		"speed": "1Gbps",
 		"state": "Requested",
 		"cloud_circuit_id": "PF-AP-LAX1-1002",
-		"account_uuid": "a2115890-ed02-4795-a6dd-c485bec3529c",
+		"account_uuid": "%s",
```
Comment (medzin) [inline internal/packetfabric/cloud_router_connection_test.go:245]: The byte slice returned by this function is not processed further by any `fmt` functions. The final value in this place will be literally `%s`.

**Review by medzin — APPROVED** [review]

**Pull request merged**